src/mainwindow.cpp

```diff
@@ -494,21 +494,53 @@ void MainWindow::applyCompanion(const Extensions::CompanionData &data)
 
     if (isUntitled() && !isTextChanged())
     {
-        QString meta = data.toMetaString();
-        meta.prepend("\n");
-        meta.append("Powered by CP Editor (https://github.com/cpeditor/cpeditor)");
+        QString comments = SettingsHelper::getCompetitiveCompanionHeadComments().replace(
+            "${time}", QDateTime::currentDateTime().toString("yyyy-MM-dd HH:mm:ss"));
```

Comment:
You can also split it into individual $time and $date.

Reply:
What about let the user set the format by `yyyy-MM-dd HH:mm:ss`?

Reply:
There are 2 options:
1. You need to add a new Preference page called "Header Comments" and let the user customize everything about Header comments.
2. Add placeholders for individual parts of date time like. $month $date $day $year $minute etc...
Whichever you feel is better.